sdk/provision/args_provisioner.go

```diff
@@ -0,0 +1,50 @@
+package provision
+
+import (
+	"context"
+
+	"github.com/1Password/shell-plugins/sdk"
+)
+
+// EnvVarProvisioner provisions secrets as command-line arguments.
+// ArgsPosition field is to control where the arguments are placed in the command line. We check for only "first" and provision those arguments first, otherwise we provision them last.
+type ArgsProvisioner struct {
+	sdk.Provisioner
+
+	Schema       map[string]sdk.FieldName
+	ArgsPosition string
```

Comment:
We could make this a map instead: `ArgsPositions map[string]int` -> represents a mapping from the arg name to the position that arg should be present in. This ties in with my other comment on `AddArgsAtIndex`

Usually it's not a good idea to use strings like "first" or "last" in order to define logic, because it's hard to discover by contributors and it's prone to errors and unhandled cases, when someone fills that in with a random string. The alternative would be to use an enum, but I think in this case, the best approach is using a structure like above to define the positions.